pkg/odo/cli/debug/portforward.go

```diff
@@ -0,0 +1,146 @@
+package debug
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+
+	componentlabels "github.com/openshift/odo/pkg/component/labels"
+	"github.com/openshift/odo/pkg/config"
+	"github.com/openshift/odo/pkg/debug"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+
+	"github.com/openshift/odo/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	k8sgenclioptions "k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+)
+
+// PortForwardOptions contains all the options for running the port-forward cli command.
+type PortForwardOptions struct {
+	Namespace string
+	Address   []string
+	PortPair  string
+
+	localPort  int
+	contextDir string
+
+	PortForwarder *debug.DefaultPortForwarder
+	StopChannel   chan struct{}
+	ReadyChannel  chan struct{}
+	*genericclioptions.Context
+	localConfigInfo *config.LocalConfigInfo
+}
```

Comment:
Should we have more comments here or is it just me being paranoid? For example, what's the `Address` supposed to mean, what's the `PortPair` supposed to look like, etc.?

Reply:
resolved